### PR TITLE
Add support for sdcard

### PR DESCRIPTION
**Pull request by gschwaer**
Hi,
I made the Sdcard work using [embedded-sdmmc](https://crates.io/crates/embedded-sdmmc). The provided example will print the file names and sizes on the first partition of the Sdcard to the serial terminal. Additionally, if a file named "SDTST.TXT" is present it will write some test data in a file called "SDTST.CSV" and read it back to compare.

The output will look like this:
```
Initializing SD card ... OK
SD Card Capacity: 15577 MB
Partition 0:
    0B  DATA
   34B  TEST.TXT
    0B  TESTDI~1
    0B  READON~1
    0B  SDTST.TXT
Write test: Success (Wrote 10 bytes)
Read test: Success (Read same 10 bytes)
Done
```

Unfortunately it pulls quite a lot of code.

Example      | Size
-----------------|--------------
blinky           | 9504
hello_world  | 8912
display         | 22564
ferris            | 32584
sdcard_test | **70028**

I tested it on a Longan Nano with a GD32VF103CB (128K Flash, 32K RAM). I'm not on an advanced Rust level yet, so please let me know if there are some ways to improve the code (and thus my understanding of Rust).

Possible improvements:
- In is mentioned in SdMmcSpi that the speed of the SPI could be increased after the init, but I couldn't find a way. So right now the SPI operates at 300kHz the whole time.
- Currently the timestamp for write operations is fixed to zero because we don't have a system time and date.

Thanks for this awesome repo,
Gero

Edit: Changed description to match the code after the second commit.

### CODE REVIEW

**Comment by gschwaer** [comment]
i added a test for reading (works) and made the example code more straight forward using less `if let` and more `unwrap`. I saw that `panic!` prints nothing, but just freezes. Is this intentional?

**Comment by gschwaer** [comment]
I realized that consuming the whole `gpiob` makes the feature incompatible with for example the display feature, so I changed the layout to only consume the gpio ports used (similar to how the lcd.rs module does things). Also I made the SPI frequency a parameter to the `configure` call. For my sdcard the `Fast` mode works.
I wrote a small example combining the sdcard and the lcd feature to display a running ferris. The code can be found [here](https://github.com/gschwaer/longan-nano/tree/dev/more_ferris).

File: src/lib.rs [hunk 1]
```diff
@@ -10,3 +10,4 @@ pub use gd32vf103xx_hal as hal;
 pub mod lcd;
 pub mod led;
 pub mod stdout;
+pub mod sdcard;
```
Comment (Disasm) [inline src/lib.rs:13]:
```suggestion
#[cfg(feature = "sdcard")]
#[cfg_attr(docsrs, doc(cfg(feature = "sdcard")))]
pub mod sdcard;
```

Comment (gschwaer) [reply src/lib.rs:15]: I missed that. Thanks for the review!

**Review by Disasm — APPROVED** [review]
Thank you!

**Pull request merged**